src/emulator/auth/widget_ui.ts

```diff
@@ -2,8 +2,23 @@
 // features as much as possible in the page below.
 
 const SCRIPT = `
+function assert(condition, error) {
+  if (!condition) {
+    if (!(error instanceof Error)) {
+      error = new Error('Auth Emulator Internal Error: ' + error);
+    }
+    // Show error with great visibility AND stops further user interactions.
+    document.body.textContent = error.stack || error.message;
+    document.body.style = 'color:red;white-space:pre';
```

Comment:
I think if you're making the background red, you should make the text white

Reply:
This only make the text red :)